### DISCUSSION TITLE
`@nxrocks/nx-spring-boot:serve` uses project `.env`, `./gradlew bootRun` does not

### DISCUSSION THREAD

**Discussion by tschaffter** [tinesoft/nxrocks#132] [General]
I'm migrating Java projects from Maven to Gradle in the Nx monorepo [Sage-Bionetworks/challenge-registry](https://github.com/Sage-Bionetworks/challenge-registry). I'm new to Gradle but the transition so far has been smooth minus one issue described in detailed in [this post](https://github.com/Sage-Bionetworks/challenge-registry/pull/578#issuecomment-1242519446).

To summarize, an app has a file `.env` that includes variables used in the Spring Boot app file `application.yml`. When starting the app with `nx serve <app-name>`, the app successfully starts, i.e. the values defined in `.env` are successfully accessed by the app.

When I move to the app folder and run `./gradlew bootRun`, which is the command run under the hood by `nx serve <app-name>`, the app fails to start because the values defined in `.env` are not used by the app. Starting the app with `DB_URL=<value> ./gradlew bootRun` works.

What could be the difference between running `nx serve <app-name>` from the workspace root folder and running `./gradlew bootRun` from the app folder?

Thanks for your help!

**Comment by tinesoft** [comment]
Hi Thomas,

the response to your question is here: https://nx.dev/recipe/define-environment-variables

Indeed, it is Nx that loads the `.env` file and makes the variables it contains, available as environment variables.

This is not something gradle wrapper (nor maven wrapper)  do by default (not their job), thus the need to combine them with another command

**Reply by tschaffter** [reply]
Thanks Tine, this makes sense!
